### ISSUE TITLE
Go completion broken when MarkPersistentRequiredFlag() called on root command

### ISSUE THREAD

**Issue by marckhouzam** [spf13/cobra#1093]
While working on supporting required flags for Go completions (Fish and Zsh v2) I hit the following bug.

If a persistent flag on the root command is marked as required, e.g., 
```
rootCmd.PersistentFlags().String("flag", "", "required persistent")
rootCmd.MarkPersistentFlagRequired("flag")
```
it breaks the `__complete` command.

This is because the `__complete` command, being a child of the root command, inherits the required persistent flag, and since that flag is not specified when calling `__complete`, the `__complete` command fails to execute.
```
$ ./testprog __complete ""
Error: required flag(s) "flag" not set
```
I think it is and unlikely case though; if the root command would have a required persistent flag, it would imply that that flag would need to be specified in every single case of calling the program; I have trouble imagining a use case for that.

Still, it should probably be fixed.
I don't have a solution right now, but I will be investigating if defining a `PreRun` on `__complete`  can allow us to turn off required flags for `__complete`.

In the unlikely case of this bug, the impacts are:
- bash: Go custom completions (`ValidArgsFunction` and `RegisterFlagCompletionFunc()`) would not work (but everything else would)
- fish: all completions would not work

**Comment by marckhouzam** [comment]
This program shows the issue:
```
package main

import (
	"fmt"

	"github.com/spf13/cobra"
)

var rootCmd = &cobra.Command{
	Args: cobra.ExactArgs(1),
	Use:  "testprog",
	Run: func(cmd *cobra.Command, args []string) {
		fmt.Println("rootCmd called")
	},
}

func main() {
	rootCmd.PersistentFlags().String("flag", "", "required persistent")
	// THIS BREAKS __complete BECAUSE IT INHERITS THIS FLAG FROM THE ROOT COMMAND
        // AND THEN validateRequiredFlags() FAILS
	rootCmd.MarkPersistentFlagRequired("flag")

	rootCmd.Execute()
}
```
and then calling
```
$ ./testprog __complete ""
Error: required flag(s) "flag" not set
```

**Comment by marckhouzam** [comment]
This problem turns out to be wider than I originally thought.

If a persistent flag on the root command is marked as required, it breaks any sub-command that uses the `DisableFlagParsing` field.

The good news is, this should be easy to fix, and will fix the `__complete` command at the same time.  I will post a PR soon.

**Comment by marckhouzam** [comment]
Fix posted in #1095
